image-rs/src/verity.rs

```diff
@@ -0,0 +1,102 @@
+use anyhow::Result;
+use base64;
+use devicemapper::{DevId, DmFlags, DmName, DmOptions, DM};
+use serde::{Deserialize, Serialize};
+use serde_json;
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct DmVerityOption {
+    /// Hash algorithm for dm-verity.
+    pub hashtype: String,
+    /// Used block size for the data device.
+    pub blocksize: String,
+    /// Used block size for the hash device.
+    pub hashsize: String,
+    /// Size of data device used in verification.
+    pub blocknum: String,
+    /// Offset of hash area/superblock on hash_device.
+    pub offset: String,
+    /// Root hash for device verification or activation.
+    pub hash: String,
+}
+
+/// Creates a mapping with <name> backed by data_device <source_device_path>
+/// and using hash_device for in-kernel verification.
+/// It will return the verity block device Path "/dev/mapper/<name>"
+/// Notes: the data device and the hash device are the same one.
+pub fn create_verity_device(
+    verity_option: &DmVerityOption,
+    source_device_path: &str,
+) -> Result<String> {
+    let dm = DM::new()?;
+    let verity_name = DmName::new(&verity_option.hash)?;
+    let id = DevId::Name(verity_name);
+    let opts = DmOptions::default().set_flags(DmFlags::DM_READONLY);
+    let device_num_blocks = verity_option.blocknum.parse::<u64>().unwrap();
```

Comment:
Please return error instead of `unwrap()` here.

Reply:
The data source is untrusted and unverified, so we can't unwrap() here to avoid possible attack.

Reply:
Thanks, I'll update it.